distrib_rl/Distrib/RedisServer.py

```diff
@@ -165,6 +176,15 @@ def _update_sps(self, collected_timesteps):
             self.accumulated_sps = 0
             self.last_sps_measure = time.time()
 
+    def _atomic_pop_all(self, key):
+        pipe = self.redis.pipeline()
+        pipe.lrange(key, 0, -1)
+        pipe.delete(key)
+        packed_results = pipe.execute()[0]
+        if packed_results is None:
+            return []
+        return [self._message_serializer.unpack(packed_result) for packed_result in packed_results]
```

Comment:
Should this share an impl with `RedisClient`?

Reply:
Left as-is for now. can clean this up more later as we better encapsulate the client logic